### PR TITLE
Added useWindowResize hook ready to be merged 

### PR DESCRIPTION
**Pull request by WJWang**
I added a  window resize hook to detect window resize event.

### CODE REVIEW

File: src/useWindowResize/index.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+
+const useWindowResize = (): [
+  number,
+  number,
+] => {
+  const [height, setHeight] = React.useState(window.innerHeight);
+  const [width, setWidth] = React.useState(window.innerWidth);
+
+  const listener = () => {
```
Comment (sandiiarov) [inline src/useWindowResize/index.tsx]:
```suggestion
  const resize = () => {
```

**Comment by sandiiarov** [comment]
Looks good! Can you add test and info about this hook to `README.md`?

File: docs/useWindowResize.mdx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  );
+};
+```
+
+<Playground>
+  {() => {
+    const [height, width] = useWindowResize();
+    return (
+      <div>
+        Currently window size : { 'height is : ' + height } and { 'width is : ' + width }
```
Comment (sandiiarov) [inline docs/useWindowResize.mdx]:
```suggestion
        <b>height:</b> {height} <b>width:</b> {width}
```

File: src/useWindowResize/index.tsx (hunk 1 above)
Comment (WJWang) [reply src/useWindowResize/index.tsx]: Fixed.

File: docs/useWindowResize.mdx (hunk 2 above)
Comment (WJWang) [reply docs/useWindowResize.mdx]: Fixed.

**Comment by sandiiarov** [comment]
@WJWang I think we can merge it as soon as you add a test.

**Comment by WJWang** [comment]
@sandiiarov I have added the test.

File: __tests__/useWindowResize.test.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    };
+
+    const { container, rerender } = render(<TestComponent />);
+    fireResize(800, 600);
+    expect(container.firstChild.textContent).toBe('800x600');
+    fireResize(1024, 768);
+    expect(container.firstChild.textContent).toBe('1024x768');
+    fireResize(1440, 800);
+    expect(container.firstChild.textContent).toBe('1440x800');
+});
```
Comment (sandiiarov) [inline __tests__/useWindowResize.test.tsx]: 2 things:
1. Can you wrap `fireResize` into the `act`?
2. Add `@ts-ignore` before 
```js
window.innerWidth = width;
window.innerHeight = height;
```

Comment (sandiiarov) [reply __tests__/useWindowResize.test.tsx]: Also you are using wrong `tab-width`

This is fixed version 👇 

```js
import * as React from 'react';
import { act, render } from 'react-testing-library';
import { useWindowResize } from '../src';

test('useWindowResize should react on window resize events', () => {
  function fireResize(width: number, height: number) {
    // @ts-ignore
    window.innerWidth = width;
    // @ts-ignore
    window.innerHeight = height;
    window.dispatchEvent(new Event('resize'));
  }

  const TestComponent = () => {
    const [height, width] = useWindowResize();
    return <span>{`${width}x${height}`}</span>;
  };

  const { container, rerender } = render(<TestComponent />);
  act(() => fireResize(800, 600));
  expect(container.firstChild.textContent).toBe('800x600');
  act(() => fireResize(1024, 768));
  expect(container.firstChild.textContent).toBe('1024x768');
  act(() => fireResize(1440, 800));
  expect(container.firstChild.textContent).toBe('1440x800');
});
```

File: src/useWindowResize/index.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    window.addEventListener('resize', resize);
+    return () => {
+      window.removeEventListener('resize', resize);
+    };
+  }, []);
+
+  return [height, width];
+};
+
+export default useWindowResize;
```
Comment (sandiiarov) [inline src/useWindowResize/index.tsx]: Prettified version 👇 
```js
import * as React from 'react';

const useWindowResize = (): [number, number] => {
  const [height, setHeight] = React.useState(window.innerHeight);
  const [width, setWidth] = React.useState(window.innerWidth);

  const resize = () => {
    setWidth(window.innerWidth);
    setHeight(window.innerHeight);
  };

  React.useEffect(() => {
    window.addEventListener('resize', resize);

    return () => {
      window.removeEventListener('resize', resize);
    };
  }, []);

  return [height, width];
};

export default useWindowResize;
```

**Comment by sandiiarov** [comment]
Sorry for the delay! Just didn't have time to look at this. So, you can use my prettified suggestions and I will merge it!

**Comment by WJWang** [comment]
Thanks for reviewing, I have changed the wrong indentation

**Pull request merged**